### PR TITLE
Deployment expiration

### PR DESCRIPTION
**Pull request by keymone**
- add config option expire_after_days
- remove deployments where .git_revision file mtime is too old
- do not deploy branches where latest commit is too old
- add specs

### CODE REVIEW

File: agent/puppetupdate.rb [hunk 1]
```diff
@@ -128,6 +130,8 @@ def resolve(git, env, limit=nil)
         Log.info "Resolving changes"
 
         changes = []
+        expiration_time = expire_after_days > 0 ?
```
Comment (tub) [inline agent/puppetupdate.rb]: You could probably pre-calculate this in the init. Not a huge deal though.

File: agent/puppetupdate.rb [hunk 2]
```diff
@@ -205,9 +216,15 @@ def resolve(git, env, limit=nil)
                 changes << msg
               end
             else
-              msg = reset_ref(ref, sha)
-              Log.info msg
-              changes << msg
+              time = run(git_cmd('show --quiet --format=format:%%at %s', ref)).to_i
+
+              if time < expiration_time
+                Log.info "ignoring #{dir}: older than #{expire_after_days} days"
```
Comment (tub) [inline agent/puppetupdate.rb]: I may be reading this wrong, but I think this log line is confusing. Isn't it ignoring `dir` because it's younger than `expire_after_days`?

Comment (keymone) [inline agent/puppetupdate.rb]: `expiration_time` is timestamp for `Time.now - 30.days`. if `time < expiration_time` then it's more than 30 days in the past and therefore "older". am i missing something?

Comment (tub) [inline agent/puppetupdate.rb]: Don't worry, I _was_ reading it wrong. It's hard to follow the script until you've read most of it through :)
